earthly

```diff
@@ -44,6 +44,10 @@ detect_frontend() {
 frontend_bin=$(detect_frontend)
 
 get_latest_binary() {
+    if [ "$frontend_bin" == "none" ]; then
+        echo "Could not find a frontend, perhaps the daemon isn't running?"
+        false
```

Comment:
should this be an explicit `exit 1` instead?

Ideally we'll have `set -e` enabled, but relying on that and using `false` to exit seems fragile.

Reply:
`exit 1` would be reasonable, or some other number. I didn't pick `exit 1` because I didn't want to overly prescribe the exit code - that's really a project choice (maybe you prefer 7...).

(In my own code I'd definitely use an explicit exit code for a thing like this)